pkg/plugins/runtime/k8s/controllers/pod_converter_test.go

```diff
@@ -26,6 +27,54 @@ import (
 	util_yaml "github.com/kumahq/kuma/pkg/util/yaml"
 )
 
+var ParseReplicaSets = func(values []string) ([]*kube_apps.ReplicaSet, error) {
+	rsets := make([]*kube_apps.ReplicaSet, len(values))
+	for i, value := range values {
+		rset := kube_apps.ReplicaSet{}
+		if err := yaml.Unmarshal([]byte(value), &rset); err != nil {
+			return nil, err
+		}
+		rsets[i] = &rset
+	}
+	return rsets, nil
+}
+
+var ParseJobs = func(values []string) ([]*kube_batch.Job, error) {
+	jobs := make([]*kube_batch.Job, len(values))
+	for i, value := range values {
+		job := kube_batch.Job{}
+		if err := yaml.Unmarshal([]byte(value), &job); err != nil {
+			return nil, err
+		}
+		jobs[i] = &job
+	}
+	return jobs, nil
```

Comment:
could this be a generic function? :thinking:

Reply:
...or just pass `[]kube_batch.Job` as a second argument

Reply:
I guess but IMO a terrible pattern that only exists because go didn't have generics.

Reply:
I agree although it sounds more idiomatic. Maybe this will change in the next years of Go with generics.

Anyways, I'm fine either way.

Reply:
I've introduced a generic function. You can take a look if that is correct.